core/src/main/scala/spire/math/ULong.scala

```diff
@@ -64,9 +64,13 @@ class ULong(val signed: Long) extends AnyVal {
     if (signed >= 0L) signed.toString
     else (ULong.LimitAsBigInt + signed).toString
 
+  // TODO: replace these with `===` and `=!=`? Don't we get `==` and `!=` for free with value classes?
```

Comment:
So, you can definitely pass anything to `==` due to Scala's support for Java's universal `.equals` method. The issue about defining the method like this has to do with the bytecode produced.

Here's a gist that look at `==` between value classes versus other types: https://gist.github.com/non/aacc4a86ec5971faea7c

As you can see, the `UByte == UByte` case is handled with a special method (located in the `UByte companion`) called `$eq$eq$extension`. Our method definition is "overriding" that one (maybe unnecessarily).

We do get these for free, so I think they can be removed, but I just wanted to point out the machinery being used.